### ISSUE TITLE
Upgrade to 16.0 fails with MySQL error

### ISSUE THREAD

**Issue by smiddy** [nextcloud/server#15511]
### Steps to reproduce
1. Have an install of Nextcloud 15
2. Use web upgrader to get to upgrade
3. After installation and disabling maintenance mode, the upgrade process to get the upgrade process fails

### Expected behaviour
Normal upgrade to 16.0

### Actual behaviour
Error: (The entry `nc_6840_collres_collections` is NOT in my database!)

`InvalidArgumentException: Primary index name on "nc_6840_collres_collections" is too long.`

Full log
```
Update vorbereiten

Log-Level auf "debug" gesetzt

Wartungsmodus eingeschaltet

Reparaturschritt: Repair MySQL collation

Reparaturinformation: All tables already have the correct collation -> nothing to do

Reparaturschritt: Repair SQLite autoincrement

Reparaturschritt: Copy data from accounts table when migrating from ownCloud

Reparaturschritt: Drop account terms table when migrating from ownCloud

Das Datenbankschema wird aktualisiert

InvalidArgumentException: Primary index name on "nc_6840_collres_collections" is too long.
```

### Server configuration

manitu.de webhosting

**Operating system**:

**Web server:**

**Database:** mysql

**PHP version:** 7.2

**Nextcloud version:** 15

**Updated from an older Nextcloud/ownCloud or fresh install:** Upgrade

**Where did you install Nextcloud from:** Automatic installer from manitu.de

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
<?php
$CONFIG = array (
  'instanceid' => ,
  'passwordsalt' => ,
  'secret' => ,
  'trusted_domains' => 
  array (
    0 => 'nextcloud.schmidt-mail.eu',
  ),
  'datadirectory' => '/home/sites/site16/web/nextcloud/data',
  'overwrite.cli.url' => 'http://www.schmidt-mail.eu/nextcloud',
  'dbtype' => 'mysql',
  'version' => '15.0.5.3',
  'dbname' => ,
  'dbhost' => 'mysql03.manitu.net',
  'dbport' => '',
  'dbtableprefix' => 'nc_6840_',
  'dbuser' => ,
  'dbpassword' => ,
  'installed' => true,
  'maintenance' => false,
  'theme' => '',
  'loglevel' => 2,
  'mail_from_address' => 'noreply',
  'mail_smtpmode' => 'smtp',
  'mail_domain' => 'nextcloud.schmidt-mail.eu',
  'mail_smtpsecure' => 'ssl',
  'mail_sendmailmode' => 'smtp',
  'mail_smtpauth' => 1,
  'mail_smtphost' => 'mail.manitu.de',
  'mail_smtpport' => '465',
  'mail_smtpname' => ,
  'mail_smtppassword' => ,
  'mail_smtpauthtype' => 'LOGIN',
  'updater.secret' =>,
);

```
</details>

**Are you using external storage, if yes which one:** none

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Firefox 66.0.4

**Operating system:** xubuntu 18.04

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

**Comment by lukstein** [comment]
I have the same problem. Same host and same error.  
Did you try to contact manitu?

**Comment by kesselb** [comment]
Duplicate of #15463

A proper fix will be part of Nextcloud 16.0.1. In the meantime you can apply the patch from https://github.com/nextcloud/server/pull/15289/files
